### PR TITLE
JIRA: VZ-9169 create Rancher credential from auth CAPI credential

### PR DESCRIPTION
**Pull request by jmaron99**
This PR addresses the need to create a Rancher OCI credential from an existing CAPI auth credential during Rancher installation (creation of cattle-global-data NS).

There needs to be another PR to complete the JIRA to address to create the CAPI secret from the Rancher credential in the postInstall phase of the CAPI component installation.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go [hunk 1]
```diff
@@ -519,6 +522,111 @@ func TestInstall(t *testing.T) {
 	}
 }
 
+func TestCreateAdminCloudCredentialSecret(t *testing.T) {
+	//userList.SetGroupVersionKind(schema.GroupVersionKind{
```
Comment (anders-swanson) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:526]: remove comments

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go [hunk 2]
```diff
@@ -519,6 +522,111 @@ func TestInstall(t *testing.T) {
...
+metadata:
+  resourceVersion: ""
+`
+	userListMap := make(map[string]interface{})
+	err := yaml.Unmarshal([]byte(userListYaml), &userListMap)
+	assert.NoError(t, err)
+	userList := &unstructured.UnstructuredList{}
+	userList.SetUnstructuredContent(userListMap)
+
+	// NOTE: Had to resort to leveraging mocks since deep in the reflection layers of the fake client creation the
```
Comment (anders-swanson) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:554]: can you use a fake client, add a single user, and then get by label selector? Not sure why you need the unstructured list.

File: platform-operator/controllers/verrazzano/reconcile/controller.go [hunk 3]
```diff
@@ -901,6 +906,31 @@ func (r *Reconciler) watchThanosInternalUserSecret(namespace string, name string
 	)
 }
 
+func (r *Reconciler) watchRancherGlobalDataNamespace(namespace string, name string) error {
```
Comment (anders-swanson) [inline platform-operator/controllers/verrazzano/reconcile/controller.go:909]: I think we might need to reconcile rancher if the CAPI secret is created

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go (hunk 2 above)
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:554]: I'd rather not introduce the go.mod dependency on the rancher lib to create a user - it isn't used anywhere else as far as I can tell.

**Review by anders-swanson — APPROVED** [review]

File: platform-operator/controllers/verrazzano/reconcile/controller.go (hunk 3 above)
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/reconcile/controller.go:909]: We discussed and put in the JIRA handling that case with a CAPI component post install

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go (hunk 2 above)
Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:549]: something like: 
```
fakeClient := fake.NewClientBuilder().WithScheme(k8scheme.Scheme).WithObjects(
		&unstructured.Unstructured{ //user data },
)
```

**Review by anders-swanson — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go (hunk 2 above)
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:549]: May work - haven't tried.  Not sure whether the fake client's list implementation is sophisticated enough to return a list with a single registered object.  I assumed a LIst() would need a formed list to return.

**Pull request merged**